packages/template-retail-react-app/app/commerce-api/hooks/useBasket.js

```diff
@@ -115,9 +105,20 @@ export default function useBasket(opts = {}) {
              * @param {number} item.quantity - The quantity of the item.
              */
             async addItemToBasket(item) {
+                let newBasket = {}
+                if (!basket || !basket.basketId) {
+                    // Back to using ShopperBaskets for all basket interaction.
+                    newBasket = await api.shopperBaskets.createBasket({})
+
+                    // Throw if there was a problem creating the basket
+                    if (isError(basket)) {
+                        throw new Error(basket)
```

Comment:
Nit: Should we pass the response error through the Error initialzation instead of the entire basket object? 🤔  or can we use custom message here. 
```
throw new Error ('Can't create basket')
```

Reply:
Yeah that would help. But since I reverted the changes. This is now out of scope of change surface.
I don't really want to include this in the upgrade guide just for an error optimization.